### ISSUE TITLE
Global proxy with secondary order authentication

### ISSUE THREAD

**Issue by cwgoes** [wyvernprotocol/wyvern-v3#32]
Goal is to allow easy `transferFrom` usage without deployment of a second proxy.

- Exchange contract requires known `AuthenticatedProxy` implementation for semantic correctness of `proxy`
  - Could add an extra, specially authorised, revised contract which has logic to check `transferFrom` signatures
  - Need to pass additional data to the proxy contract so that it can authenticate signatures
- Alternative approach: make order maker a contract
  - Use the same proxy contract
  - `isValidSignature`-equivalent method checks signature on behalf of user
    - Basically checks that the user (looked up from call data) signed the Wyvern order
    - Is there an altered version of EIP 1271 that allows passing extra data?
  - Needs to pass call data as well so user can be looked up
  - Users then approve proxy deployed by this contract

**Comment by cwgoes** [comment]
![42b1e9ed-8a08-4796-981d-7a6dfd05c683](https://user-images.githubusercontent.com/24760279/99451574-ad6a8c00-2922-11eb-8602-861c0abb4246.jpeg)

**Comment by cwgoes** [comment]
Supports interface for contract calls, first try special one, then try EIP 1271 one.
